deepchecks/vision/datasets/classification/mnist.py

```diff
@@ -205,9 +215,111 @@ def __call__(self, batch):
         return torch.stack(results)
 
 
-class MNIST(datasets.MNIST):
+class IterableTorchMnistDataset(IterableDataset):
+    """Iterable MNIST <http://yann.lecun.com/exdb/mnist/>`_ Dataset.
+
+    Parameters
+    ----------
+    batch_size: int, default=64
+        Batch size to use
+    train: bool, default: true
+        If True, creates dataset from ``training.pt``, otherwise from ``test.pt``.
+    transform: t.Optional[t.Callable], default: None
+        A function/transform that  takes in an PIL image and returns a transformed version.
+        E.g, ``transforms.RandomCrop``
+    n_samples: int, default: None
+        Number of samples to use. If None, use all samples.
+    """
+
+    mirrors = [
+        'http://yann.lecun.com/exdb/mnist/',
+        'https://ossci-datasets.s3.amazonaws.com/mnist/',
+    ]
+
+    resources = [
+        ('train-images-idx3-ubyte.gz', 'f68b3c2dcbeaaa9fbdd348bbdeb94873'),
+        ('train-labels-idx1-ubyte.gz', 'd53e105ee54ea40749a09fcbcd1e9432'),
+        ('t10k-images-idx3-ubyte.gz', '9fb629c4189551a2d022fa330f9573f3'),
+        ('t10k-labels-idx1-ubyte.gz', 'ec29112dd5afa0611ce80d1b7f02629c')]
+
+    def __init__(self, batch_size: int = 64, train: bool = True, transform: t.Optional[t.Callable] = None,
+                 n_samples: int = None) -> None:
+        super().__init__()
+        self.train = train  # training set or test set
+        self.transform = transform
+        self.batch_size = batch_size
+
+        self.download()
+        if not self._check_exists():
+            raise RuntimeError('Dataset not found.' +
+                               ' You can use download=True to download it')
+
+        self.data, self.targets = self._load_data()
+        if n_samples:
+            self.data = self.data[:n_samples]
+            self.targets = self.targets[:n_samples]
+        if self.transform is not None:
+            self.data = torch.stack([self.transform(image=img.numpy())['image'] for img in self.data])
```

Comment:
why not .cpu.detach?

Reply:
use object_to_numpy

Reply:
it can either cpu or gpu and we are fine with it (it simulates a user dataset)

Reply:
ok